vision/unit_tests/test_module.py

```diff
@@ -1,94 +1,200 @@
+            pass
+
+        ## Empty image
+        with self.subTest(i="Empty Image"):
+            pass
+
+        ## None as image
+        with self.subTest(i="None as Image"):
+            pass
+
+    def test_return(self):
+        """
+        Verify returns only expected output types.
+
+        Returns
+        -------
+        bool
+        """
+        ##
+        for i in range(1, 6):
+            color_image = 255 * np.ones((i * 100, i * 100, 3), dtype="uint8")
+            color_image = cv2.circle(color_image, (i * 50, i * 50), 20, (0, 0, 0), 4)
+
+            result = mif(color_image, None)
+
+            self.assertIn(result, [True, False])
+
+        ## Ensure does not modify original image
+        color_image = 255 * np.ones((100, 100, 3), dtype="uint8")
+        color_image = cv2.circle(color_image, (50, 50), 20, (0, 0, 0), 4)
+
+        color_parameter = np.copy(color_image)
+
+        mif(color_parameter, None)
+
+        np.testing.assert_array_equal(color_image, color_parameter)
+
+
+class TestModuleOrientation(unittest.TestCase):
+    """
+    Testing module.get_module_orientation for validity.
+    """
+
+    def test_params(self):
+        """
+        Verify can handle range of input type.
+
+        Parameters
+        ----------
+        roi: ndarray
+            module region of interest calculated by region_of_interest
+        """
+        IMAGE_SIZE = [1920, 1080]
+
+        # testing with ndarray of all zeroes
+        image = np.zeros(IMAGE_SIZE)
+        result = get_module_orientation(image)
+
+        self.assertIs(type(image), np.ndarray)
+        self.assertIs(type(result), tuple)
+
+    def test_get_module_orientation(self):
+
+        img_dir = os.path.join(gparent_dir, "vision_images/module/Feb29")
+        files = os.listdir(img_dir)
+
+        # hand-picked images for unit test, assumes color-depth pair for each of the files
+        estimates = {
+            # FORMAT: "name_of_file_without_color-depth_tail" : [estimated_center, estimated_orientation]
+            "2020-02-29_15.35.31.313411": [(906, 807), (-25, -1)],
+            "2020-02-29_15.35.40.045632": [(641, 866), (-70, -25)],
+            "2020-02-29_15.35.57.551331": [(1228, 653), (0, 5)],
+            "2020-02-29_15.35.56.436446": [(1170, 644), (10, 10)],
+            "2020-02-29_15.40.42.462549": [(948, 620), (20, 5)]
+            # NOTE: The center tuples are estimated because center function does not work at the moment
+            # TODO: Find more head-on shots of the module
+        }
+
+        counter = 0
+        for current_file in files:
+            counter += 1  # started one ahead to get next (depth) file
+            stripped_filename = current_file[:26]
+
+            # Get files two at a time, color w/ depth
+            if (
+                counter - 1
+            ) % 2 == 0 and stripped_filename in estimates:  # counts every other file and checks if current is in estimates

```

Comment:
> if (
                counter - 1
            ) % 2 == 0 and stripped_filename in estimates:

For this here, if you just loop through the files you have in estimates, you can remove the extra logic with the counter and checking if the file is in estimates.